boards/FSM/Src/Io/Io_Scancon2RMHF.c

```diff
@@ -0,0 +1,47 @@
+#include <assert.h>
+#include <stdlib.h>
+#include "Io_Scancon2RMHF.h"
+
+struct Encoder
+{
+    TIM_HandleTypeDef *htim;
+};
+
+static struct Encoder primary_encoder, secondary_encoder;
+
+void Io_Scancon2RMHF_Init(
+    TIM_HandleTypeDef *const htim_primary_encoder,
+    TIM_HandleTypeDef *const htim_secondary_encoder)
+{
+    assert(htim_primary_encoder != NULL);
+    assert(htim_secondary_encoder != NULL);
+
+    primary_encoder.htim   = htim_primary_encoder;
+    secondary_encoder.htim = htim_secondary_encoder;
+
+    assert(
+        HAL_TIM_Encoder_Start(primary_encoder.htim, TIM_CHANNEL_ALL) != HAL_OK);
+    assert(
+        HAL_TIM_Encoder_Start(htim_secondary_encoder, TIM_CHANNEL_ALL) !=
+        HAL_OK);
```

Comment:
Hmm, I think we want to assert `== HAL_OK` instead for both of these here?

Reply:
Might want to file a ticket to actually test this code out with hardware if you can't access hardware right now.

Reply:
asserts should be fixed